test/application.js

```diff
@@ -412,9 +412,12 @@ describe('app.respond', function(){
         this.status = 200;
         res.setHeader("Content-Type", "text/html")
         res.write('Hello');
-        setTimeout(function(){
-          res.end("Goodbye")
-        }, 0);
+        yield new Promise(function(resolve){
+          setTimeout(function(){
+            res.end("Goodbye");
+            resolve();
+          }, 0);
+        });
```

Comment:
This is necessary because before `respond` wouldn't do anything. Now we have to make sure that it doesn't close the socket too soon.